src/main/java/io/fabric8/jenkins/openshiftsync/PodTemplateUtils.java

```diff
@@ -423,7 +424,8 @@ protected static boolean configMapContainsSlave(ConfigMap configMap) {
 
     protected static boolean hasSlaveLabelOrAnnotation(Map<String, String> map) {
         return map != null && map.containsKey(IMAGESTREAM_AGENT_LABEL)
-                && map.get(IMAGESTREAM_AGENT_LABEL).equals(IMAGESTREAM_AGENT_LABEL_VALUE);
+                && (map.get(IMAGESTREAM_AGENT_LABEL).equals(LEGACY_IMAGESTREAM_AGENT_LABEL_VALUE) ||
```

Comment:
for null protection, can we use 
```
LEGACY_IMAGESTREAM_AGENT_LABEL_VALUE.equals(map.get(IMAGESTREAM_AGENT_LABEL)) 
```
and extract `map.get(IMAGESTREAM_AGENT_LABEL)`  into a variable.
?

Reply:
Sure thing.  Will work on that shortly.

Reply:
update pushed